src/chaincode/dex/manageWhitelist.spec.ts

```diff
@@ -0,0 +1,338 @@
+    const currencyClassKey: TokenClassKey = currency.tokenClassKey();
+    const currencyBalance: TokenBalance = currency.tokenBalance();
+
+    const dexInstance: TokenInstance = dexTestUtils.tokenInstance();
+    const dexClass: TokenClass = dexTestUtils.tokenClass();
+    const dexClassKey: TokenClassKey = dexTestUtils.tokenClassKey();
+    const dexBalance: TokenBalance = dexTestUtils.tokenBalance();
+
+    const dexFeeConfig: DexFeeConfig = new DexFeeConfig([asValidUserAlias(users.admin.identityKey)], 0.1);
+
+    const { ctx, contract } = fixture<GalaChainContext, DexV3Contract>(DexV3Contract)
+      .registeredUsers(users.testUser1, users.testUser2)
+      .savedState(
+        currencyInstance,
+        currencyClass,
+        currencyBalance,
+        dexFeeConfig,
+        dexInstance,
+        dexClass,
+        dexBalance
+      )
+      .savedRangeState([]);
+
+    // Create a private pool first
+    const whitelist = [users.testUser1.identityKey];
+    const createDto = new CreatePoolDto(
+      dexClassKey,
+      currencyClassKey,
+      DexFeePercentageTypes.FEE_1_PERCENT,
+      new BigNumber("1"),
+      true, // isPrivate
+      whitelist
+    );
+    createDto.uniqueKey = "test-create-1";
+    createDto.sign(users.testUser1.privateKey);
+
+    await contract.CreatePool(ctx, createDto);
+
+    // Add user to whitelist
+    const manageDto = new ManageWhitelistDto(
+      dexClassKey,
+      currencyClassKey,
+      DexFeePercentageTypes.FEE_1_PERCENT,
+      users.testUser2.identityKey,
+      true // isAdd = true
+    );
+    manageDto.uniqueKey = "test-manage-1";
+    manageDto.sign(users.testUser1.privateKey); // Whitelisted user
+
+    // When
+    const response = await contract.ManageWhitelist(ctx, manageDto);
+
+    // Then
+    expect(response).toEqual(GalaChainResponse.Success(undefined));
+  });
+
+  it("should remove user from whitelist when called by whitelisted user", async () => {
+    // Given
+    const currencyInstance: TokenInstance = currency.tokenInstance();
+    const currencyClass: TokenClass = currency.tokenClass();
+    const currencyClassKey: TokenClassKey = currency.tokenClassKey();
+    const currencyBalance: TokenBalance = currency.tokenBalance();
+
+    const dexInstance: TokenInstance = dexTestUtils.tokenInstance();
+    const dexClass: TokenClass = dexTestUtils.tokenClass();
+    const dexClassKey: TokenClassKey = dexTestUtils.tokenClassKey();
+    const dexBalance: TokenBalance = dexTestUtils.tokenBalance();
+
+    const dexFeeConfig: DexFeeConfig = new DexFeeConfig([asValidUserAlias(users.admin.identityKey)], 0.1);
+
+    const { ctx, contract } = fixture<GalaChainContext, DexV3Contract>(DexV3Contract)
+      .registeredUsers(users.testUser1, users.testUser2)
+      .savedState(
+        currencyInstance,
+        currencyClass,
+        currencyBalance,
+        dexFeeConfig,
+        dexInstance,
+        dexClass,
+        dexBalance
+      )
+      .savedRangeState([]);
```

Comment:
it's optional in tests
```suggestion
      );
```